From da85329befffa2e8ed5a87a8b8c49381278e5023 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Wed, 21 Mar 2012 15:32:55 +1000
Subject: [PATCH 3/6] Don't pass NULL to fclose(3)

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/wcmISDV4.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/wcmISDV4.c b/src/wcmISDV4.c
index 5a123cd..892fbff 100644
--- a/src/wcmISDV4.c
+++ b/src/wcmISDV4.c
@@ -989,7 +989,8 @@ model_from_sysfs(const InputInfoPtr pInfo, int fd, int *id)
 out:
 	udev_device_unref(device);
 	udev_unref(udev);
-	fclose(file);
+	if (file)
+		fclose(file);
 	free(sysfs_path);
 	return model;
 }
-- 
1.7.7.6

